policy.go

```diff
@@ -72,61 +69,128 @@ type policyOrDataTree []policyOrDataNode
+	authPublicKey          *tpm2.Public
+	pcrPolicyCounterHandle tpm2.Handle
+	v0PinIndexAuthPolicies tpm2.DigestList
+}
+
+// staticPolicyDataRaw_v0 is version 0 of the on-disk format of staticPolicyData.
+type staticPolicyDataRaw_v0 struct {
 	AuthPublicKey        *tpm2.Public
 	PinIndexHandle       tpm2.Handle
 	PinIndexAuthPolicies tpm2.DigestList
 }
 
-// staticPolicyDataRaw_v0 is version 0 of the on-disk format of staticPolicyData. They are currently the same structures.
-type staticPolicyDataRaw_v0 staticPolicyData
-
 func (d *staticPolicyDataRaw_v0) data() *staticPolicyData {
-	return (*staticPolicyData)(d)
+	return &staticPolicyData{
+		authPublicKey:          d.AuthPublicKey,
+		pcrPolicyCounterHandle: d.PinIndexHandle,
+		v0PinIndexAuthPolicies: d.PinIndexAuthPolicies}
 }
 
-// makeStaticPolicyDataRaw_v0 converts staticPolicyData to version 0 of the on-disk format. They are currently the same structures
-// so this is just a cast, but this may not be the case if the metadata version changes in the future.
+// makeStaticPolicyDataRaw_v0 converts staticPolicyData to version 0 of the on-disk format.
 func makeStaticPolicyDataRaw_v0(data *staticPolicyData) *staticPolicyDataRaw_v0 {
-	return (*staticPolicyDataRaw_v0)(data)
+	return &staticPolicyDataRaw_v0{
+		AuthPublicKey:        data.authPublicKey,
+		PinIndexHandle:       data.pcrPolicyCounterHandle,
+		PinIndexAuthPolicies: data.v0PinIndexAuthPolicies}
+}
+
+// staticPolicyDataRaw_v1 is version 1 of the on-disk format of staticPolicyData.
+type staticPolicyDataRaw_v1 struct {
+	AuthPublicKey          *tpm2.Public
+	PCRPolicyCounterHandle tpm2.Handle
+	PCRPolicyRef           tpm2.Nonce
+}
+
+func (d *staticPolicyDataRaw_v1) data() *staticPolicyData {
+	return &staticPolicyData{
+		authPublicKey:          d.AuthPublicKey,
+		pcrPolicyCounterHandle: d.PCRPolicyCounterHandle}
+}
+
+// makeStaticPolicyDataRaw_v1 converts staticPolicyData to version 1 of the on-disk format.
+func makeStaticPolicyDataRaw_v1(data *staticPolicyData) *staticPolicyDataRaw_v1 {
+	return &staticPolicyDataRaw_v1{
+		AuthPublicKey:          data.authPublicKey,
+		PCRPolicyCounterHandle: data.pcrPolicyCounterHandle}
 }
 
-// incrementDynamicPolicyCounter will increment the NV counter index associated with nvPublic. This is designed to operate on a
-// NV index created by createPinNVIndex. The authorization policy digests returned from createPinNVIndex must be supplied via the
-// nvAuthPolicies argument.
+// computePcrPolicyCounterAuthPolicies computes the authorization policy digests passed to TPM2_PolicyOR for a PCR
+// policy counter that can be updated with the key associated with updateKeyName.
+func computePcrPolicyCounterAuthPolicies(alg tpm2.HashAlgorithmId, updateKeyName tpm2.Name) (tpm2.DigestList, error) {
+	// The NV index requires 2 policies:
+	// - A policy to initialize the index with no authorization
+	// - A policy for updating the index to revoke old PCR policies, using a signed assertion.
```

Comment:
these are much simpler than the v0 policies,   a comment here like  

// These are simpler than v0 because ...

would be useful

Reply:
I've added this.